src/qat/purr/compiler/hardware_models.py

```diff
@@ -305,8 +310,11 @@ def get_devices_from_pulse_channel(self, id_: str):
     def get_pulse_channel_from_id(self, id_: str):
         return self.pulse_channels.get(id_)
 
-    def get_devices_from_physical_channel(self, id_: str):
-        physical_channel = self.get_physical_channel(id_)
+    def get_devices_from_physical_channel(
+        self, physical_channel: Union[str, PhysicalChannel]
+    ):
+        if isinstance(physical_channel, str):
+            physical_channel = self.get_physical_channel(physical_channel)
```

Comment:
the `self.get_physical_channel()` method is too short, please inline it here, no need for extra call:

```
    def get_physical_channel(self, id_: str):
        return self.physical_channels.get(id_, None)
```